zerver/views/user_settings.py

```diff
@@ -96,13 +97,14 @@ def confirm_email_change(request: HttpRequest, confirmation_key: str) -> HttpRes
         context=context,
         realm=user_profile.realm,
     )
-
+    old_email_address = Address(addr_spec=escape(old_email))
+    new_email_address = Address(addr_spec=escape(new_email))
```

Comment:
Why is `escape` called here? It very much looks like you're double-escaping.

Can you add backend test verification of this HTML using an email address where escaping will be nontrivial, and move the backend changes to an independent commit?